src/librender/mesh.cpp

```diff
@@ -107,74 +113,96 @@ MTS_VARIANT Mesh<Float, Spectrum>::Mesh(const std::string &name,
-                    //TODO: transform
-                    tmp_vertices.push_back(m_to_world.transform_affine(position));
-                    tmp_normals.push_back(normalize(m_to_world.transform_affine(normal)));
-
-                    ScalarIndex final_index = vertex_ctr++;
-                    vertex_map[vert_index] = final_index;
-                    /*TODO: LuxCore's design may create new vertices unnecessarily,
-                    since there always exists only one version of the vertex in the mapping,
-                    try to do something similar to the OBJMesh loader instead*/
-                    triangle[i] = (ScalarIndex) final_index;
-                }
-                else
-                {
-                    triangle[i] = vertex_map[vert_index];
-                }
-            }
+        const blender::MVert &v0 = verts[loops[tri_loop.tri[0]].v];
+        const blender::MVert &v1 = verts[loops[tri_loop.tri[1]].v];
+        const blender::MVert &v2 = verts[loops[tri_loop.tri[2]].v];
+
+        std::array<InputPoint3f, 3> face_points;
+        face_points[0] = InputPoint3f(v0.co[0], v0.co[1], v0.co[2]);
+        face_points[1] = InputPoint3f(v1.co[0], v1.co[1], v1.co[2]);
+        face_points[2] = InputPoint3f(v2.co[0], v2.co[1], v2.co[2]);
+
+        InputNormal3f normal;
+        if(!(blender::ME_SMOOTH & face.flag)){
+            //flat shading, use per face normals
+            const Vector3f e1 = face_points[1] - face_points[0];
+            const Vector3f e2 = face_points[2] - face_points[0];
+            normal = normalize(m_to_world.transform_affine(cross(e1, e2)));
         }
-        else{
-            //flat shading, compute and use the face normal
-			const blender::MVert &v0 = verts[loops[tri_loop.tri[0]].v];
-			const blender::MVert &v1 = verts[loops[tri_loop.tri[1]].v];
-			const blender::MVert &v2 = verts[loops[tri_loop.tri[2]].v];
-
-			const InputPoint3f p0(v0.co[0], v0.co[1], v0.co[2]);
-			const InputPoint3f p1(v1.co[0], v1.co[1], v1.co[2]);
-			const InputPoint3f p2(v2.co[0], v2.co[1], v2.co[2]);
-
-			const Vector3f e1 = p1 - p0;
-			const Vector3f e2 = p2 - p0;
-			InputNormal3f face_normal(normalize(m_to_world.transform_affine(normalize(cross(e1, e2)))));
-
-            for(u_int i=0; i<3; i++){
-                const size_t loop_index = tri_loop.tri[i];
-                const size_t vert_index = loops[loop_index].v;
-                const blender::MVert &vert = verts[vert_index];
-
-                InputPoint3f position(vert.co[0], vert.co[1], vert.co[2]);
-                //TODO: colors and UVs
-                tmp_vertices.push_back(m_to_world.transform_affine(position));
-                tmp_normals.push_back(face_normal);
-                triangle[i] = vertex_ctr++;
-                //no need for a hash map here, all vertices need to be duplicated
-                //TODO: remove duplicate vertices in case original face is not a triangle
+        for(u_int i=0; i<3; i++){
```

Comment:
`for(u_int i = 0; i < 3; i++) {`

Also `u_int` isn't used anywhere else in the codebase. Maybe could use `uint32_t` instead?